### PR TITLE
Feature/merge 4.3

### PR DESCRIPTION
**Pull request by chtyim**
All conflict resolutions are done by picking the develop branch over the release branch.

### CODE REVIEW

**Comment by chtyim** [comment]
Set of conflict files (minus **/pom.xml)

```
	both modified:   cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java
	both modified:   cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java
	both modified:   cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/DefaultStoreTest.java
	both modified:   cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java
	both modified:   cdap-distributions/bin/build_docs_bucket.sh
	both modified:   cdap-distributions/src/emr/install.sh
	both modified:   cdap-distributions/src/hdinsight/pkg/install.sh
	both modified:   cdap-distributions/src/hdinsight/pkg/mainTemplate.json
	both modified:   cdap-distributions/src/packer/cdap-sdk-ubuntu16-with-uri.json
	both modified:   cdap-distributions/src/packer/files/cdap-sdk.json
	both modified:   cdap-examples/resources/weblog-analytics-config.json
	both modified:   cdap-examples/resources/weblog-analytics.txt
	both modified:   cdap-examples/resources/weblog-service.txt
	both modified:   cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java
	both modified:   cdap-ui/app/cdap/components/FastAction/TruncateAction/index.js
	both modified:   cdap-ui/app/cdap/components/ProgramTable/index.js
	both modified:   cdap-ui/package.json
```

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-DUT6344

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java [hunk 1]
```diff
@@ -773,7 +773,6 @@ private boolean isValid(RunRecordMeta existing, byte[] sourceId, String recordTy
       LOG.debug("Current source id '{}' is not larger than the existing source id '{}' in the existing " +
                   "run record meta '{}'. Skip recording program {}.",
                 Bytes.toHexString(sourceId), Bytes.toHexString(existingSourceId), existing, recordType);
-      return false;
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java]: this is still there on develop

**Review by albertshau — COMMENTED** [review]
one comment, otherwise lgtm

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java (hunk 1 above)
Comment (chtyim) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java]: oh yeah, missing that.

**Comment by chtyim** [comment]
Build passed

**Pull request merged**